great_expectations/rule_based_profiler/util.py

```diff
@@ -1,13 +1,76 @@
 import copy
-from typing import Any, Dict, Optional, Union
+import uuid
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
 
 import great_expectations.exceptions as ge_exceptions
-from great_expectations.core.batch import BatchRequest
-from great_expectations.rule_based_profiler.domain_builder.domain import Domain
-from great_expectations.rule_based_profiler.parameter_builder.parameter_container import (
+from great_expectations import DataContext
+from great_expectations.core.batch import Batch, BatchRequest
+from great_expectations.rule_based_profiler.domain_builder import Domain
+from great_expectations.rule_based_profiler.parameter_builder import (
     ParameterContainer,
     get_parameter_value_by_fully_qualified_parameter_name,
 )
+from great_expectations.validator.validator import Validator
+
+NP_EPSILON: np.float64 = np.finfo(float).eps
+
+
+def get_validator(
+    data_context: Optional[DataContext] = None,
+    batch_request: Optional[Union[BatchRequest, dict, str]] = None,
+    domain: Optional[Domain] = None,
+    variables: Optional[ParameterContainer] = None,
+    parameters: Optional[Dict[str, ParameterContainer]] = None,
+) -> Optional[Validator]:
+    if batch_request is None:
+        return None
+
+    batch_request = build_batch_request(
+        domain=domain,
+        batch_request=batch_request,
+        variables=variables,
+        parameters=parameters,
+    )
+
+    expectation_suite_name: str = "tmp_suite"
+    if domain is None:
+        expectation_suite_name = f"{expectation_suite_name}_{str(uuid.uuid4())[:8]}"
+    else:
+        expectation_suite_name = (
+            f"{expectation_suite_name}_{domain.id}_{str(uuid.uuid4())[:8]}"
+        )
```

Comment:
Instead of this, I think that validator needs to be able to be instantiated without a suite. I thought that was possible by setting interactive mode to false. If it's not...we should make it possible.

Reply:
@jcampbell This is currently not supported by `Validator` -- the omission of the `expectation_suite_name` argument results in the `expectation_suite_name` being set to `default` in the `Validator` class (internally).  This is not ideal, because it will cause thus named expectation suite to be overwritten.  We can change the default behavior to the randomly generated `expectation_suite_name` instead of default -- would that make sense?  (By the way, interactive mode flag is not related to this situation.  If interactive mode is `False`, then we do not use the `Validator` at all, because `Validator` implies having access to the `Batch` of data.)

Reply:
Can you create a ticket to review validator design to ensure that when it is *not* in interactive mode it does *not* require/create an expectation suite?

Reply:
@jcampbell I created a ticket (`https://superconductive.atlassian.net/browse/GE-304`); however, as far as I can tell, the `interactive_evaluation` in the `Validator` deals with validation, not setting expectations (but there is no `docstring` for this parameter, hence I could be missing something, so happy to discuss so we can figure out the course of action).

Reply:
We discussed it at Architecture Review on 6/24/2021 and agreed to -- for now -- simply use `tmp.` in the randomly generated expectation suite name (filed as `JIRA-307`).